data/src/main/java/com/example/data/backendapi/service/CustomerService.java

```diff
@@ -0,0 +1,36 @@
+package com.example.data.backendapi.service;
+
+import com.example.data.backendapi.dao.CustomerRepository;
+import com.example.data.backendapi.dto.Customer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+
+import java.util.Optional;
+
+@Scope("singleton")
+public class CustomerService {
+
+    @Autowired
+    private CustomerRepository repository;
```

Comment:
Make it final. And with Lombok you just can add annotation `@RequiredArgsConstructor` so it would maintain all your final fields as an arguments for injection in the constructor. Easy to maintain!